### PR TITLE
feat(route53-targets): Add aws-route53-targets/InterfaceVpcEndpointTarget

### PR DESCRIPTION
**Pull request by KingOfPoptart**
We found a gap in route53 targets.  This new target takes in an `ec2.IInterfaceVpcEndpoint` and sets it as an alias target for Route53.

While I acknowledge the implementation isn't the prettiest, it works, is tested, and I'm all ears for improvement.

### CODE REVIEW

**Comment by flemjame-at-amazon** [comment]
Will this make it into the next revision of CDK? I've written my own workaround InterfaceVpcEndpointTarget class and I'd love to delete it from my codebase.

**Comment by KingOfPoptart** [comment]
@shivlaks Hiya - can anybody take a look at this please?

**Comment by shivlaks** [comment]
> @shivlaks Hiya - can anybody take a look at this please?

my apologies, I looked at this a long time ago and it turns out I didn't publish the comments... my bad!

File: packages/@aws-cdk/aws-route53-targets/lib/interface-vpc-endpoint-target.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+export class InterfaceVpcEndpointTarget implements r53.IAliasRecordTarget {
+  private readonly cfnVpcEndpoint: ec2.CfnVPCEndpoint;
+  constructor(private readonly vpcEndpoint: ec2.IInterfaceVpcEndpoint) {
+    this.cfnVpcEndpoint = this.vpcEndpoint.node.findChild('Resource') as ec2.CfnVPCEndpoint;
+  }
+
+  public bind(_record: r53.IRecordSet): r53.AliasRecordTargetConfig {
+    return {
+      dnsName: cdk.Fn.select(1, cdk.Fn.split(':', cdk.Fn.select(0, this.cfnVpcEndpoint.attrDnsEntries))),
+      hostedZoneId: cdk.Fn.select(0, cdk.Fn.split(':', cdk.Fn.select(0, this.cfnVpcEndpoint.attrDnsEntries))),
```
[the remark this answers is not in the document]
Comment (KingOfPoptart) [reply packages/@aws-cdk/aws-route53-targets/lib/interface-vpc-endpoint-target.ts]: So far in our testing this has worked 100% of the time.  Not the best way of implementing, but currently its the only way we've found to make this work.

[the remark this answers is not in the document]
Comment (flemjame-at-amazon) [reply packages/@aws-cdk/aws-route53-targets/lib/interface-vpc-endpoint-target.ts]: Looking at the AWS docs, the first DNS entry is the regional one, the first part of it is the hosted zone, and the second part is the dns name:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpcendpoint.html

It's not enforced for AWS marketplace services though, which doesn't guarantee any order. Maybe make a readme note that this only applies to non-marketplace services?

[the remark this answers is not in the document]
Comment (KingOfPoptart) [reply packages/@aws-cdk/aws-route53-targets/lib/interface-vpc-endpoint-target.ts]: @flemjame-at-amazon good suggestion, added

**Pull request merged**